src/function/probability/lgamma.js

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable no-loss-of-precision */
+
+import { lgammaNumber, lnSqrt2PI } from '../../plain/number/index.js'
+import { deepMap } from '../../utils/collection.js'
+import { factory } from '../../utils/factory.js'
+
+const name = 'lgamma'
+const dependencies = ['Complex', 'typed']
+
+export const createLgamma = /* #__PURE__ */ factory(name, dependencies, ({ Complex, typed }) => {
+  const TWOPI = 6.2831853071795864769252842 // 2*pi
+  const LOGPI = 1.1447298858494001741434262 // log(pi)
+  const SMALLX = 7
+  const SMALLY = 7
+
+  const REFLECTION = 0.1
+
+  const coeffs = [
+    -2.955065359477124183e-2, 6.4102564102564102564e-3, -1.9175269175269175269e-3, 8.4175084175084175084e-4,
+    -5.952380952380952381e-4, 7.9365079365079365079e-4, -2.7777777777777777778e-3, 8.3333333333333333333e-2
+  ]
+
+  /**
+   * Logarithm of the gamma function for real, positive numbers.
+   * Based off of Tom Minka's lightspeed toolbox.
+   *
+   * Syntax:
+   *
+   *    math.lgamma(n)
+   *
+   * Examples:
+   *
+   *    math.lgamma(5)       // returns 3.178053830347945
+   *    math.lgamma(0)       // returns Infinity
+   *    math.lgamma(-0.5)    // returns NaN
+   *    math.lgamma(math.i)  // returns -0.6509231993018536 - 1.8724366472624294i
+   *
+   * See also:
+   *
+   *    gamma
+   *
+   * @param {number | Complex | Array | Matrix} n   A real or complex number
+   * @return {number | Complex | Array | Matrix}    The log gamma of `n`
+   */
+
+  return typed(name, {
+    number: lgammaNumber,
+
+    Complex: function (n) {
+      if (n.isNaN()) {
+        return new Complex(NaN, NaN)
+      } else if (n.im === 0) {
+        return new Complex(this(n.re), 0)
+      } else if (n.re >= SMALLX || Math.abs(n.im) >= SMALLY) {
+        return lgammaStriling(n)
+      } else if (n.re <= REFLECTION) {
```

Comment:
While identifiers are arbitrary, I think you are using a Stirling approximation here, so it would be clearest for readers of the code if the identifier were lgammaStirling. Please change.

Reply:
Oh sorry that's a typo